### PR TITLE
[JSDoc] Add missing documentation for botbuilder-azure

### PR DESCRIPTION
**Pull request by ceciliaavila**
Addresses # 2602

## Description
This PR replaces `warn` with `error` in the `jsdoc/require-jsdoc` rule in [.eslintrc.json](https://github.com/microsoft/botbuilder-js/blob/master/.eslintrc.json#L87) file for [botbuilder-azure](https://github.com/microsoft/botbuilder-js/tree/master/libraries/botbuilder-azure) and adds all the missing documentation based on the errors thrown. 
Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet)

## Specific Changes

  - Updated `jsdoc/require-jsdoc` in `.eslintrc.json` from `warn` to `error`.
  - Added JSDoc comments in all public methods and some of the private ones. 

## Testing
The following image shows the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

File: libraries/botbuilder-azure/src/azureBlobTranscriptStore.ts [hunk 1]
```diff
@@ -323,7 +364,11 @@ export class AzureBlobTranscriptStore implements TranscriptStore {
         return AzureBlobTranscriptStore[checkedCollectionsKey][key];
     }
 
-    private createBlobService({storageAccountOrConnectionString, storageAccessKey, host}: BlobStorageSettings): BlobServiceAsync {
+    /**
+     * Create a Blob Service.
+     * @param param0 Settings required for configuring the Blob Service (storage account or connection string, access key and host).
```
Comment (santgr11) [inline libraries/botbuilder-azure/src/azureBlobTranscriptStore.ts:369]: Is this param name correct?

Comment (sw-joelmut) [reply libraries/botbuilder-azure/src/azureBlobTranscriptStore.ts:369]: I found this, https://jsdoc.app/tags-param.html#parameters-with-properties for destructuring objects in parameters 'Documenting a destructuring parameter'

File: libraries/botbuilder-azure/src/blobStorage.ts [hunk 2]
```diff
@@ -270,6 +291,13 @@ export class BlobStorage implements Storage {
         return checkedCollections[key];
     }
 
+    /**
+     * Create a Blob Service.
+     * @param storageAccountOrConnectionString Azure CloudStorageAccount instance or Connection String.
+     * @param storageAccessKey Blob Service Access Key.
+     * @param host Blob Service Host.
+     * @returns the blob services created.
```
Comment (sw-joelmut) [inline libraries/botbuilder-azure/src/blobStorage.ts:299]:
```suggestion
     * @returns The blob services created.
```

File: libraries/botbuilder-azure/src/cosmosDbPartitionedStorage.ts [hunk 3]
```diff
@@ -147,6 +147,11 @@ export class CosmosDbPartitionedStorage implements Storage {
         this.cosmosDbStorageOptions = cosmosDbStorageOptions;
     }
 
+    /**
+     * Read one or more items with matching keys from the Cosmos DB container.
+     * @param keys A collection of Ids for each item to be retrieved.
+     * @returns the read items.
```
Comment (sw-joelmut) [inline libraries/botbuilder-azure/src/cosmosDbPartitionedStorage.ts:153]:
```suggestion
     * @returns The read items.
```

File: libraries/botbuilder-azure/src/cosmosDbStorage.ts [hunk 4]
```diff
@@ -133,7 +133,12 @@ export class CosmosDbStorage implements Storage {
         this.documentCollectionCreationRequestOption = settings.documentCollectionRequestOptions;
     }
 
-    public read(keys: string[]): Promise<StoreItems> {        
+    /**
+     * Read storage items from storage.
+     * @param keys Keys of the items to read from the store.
+     * @returns the read items.
```
Comment (sw-joelmut) [inline libraries/botbuilder-azure/src/cosmosDbStorage.ts:139]:
```suggestion
     * @returns The read items.
```

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

**Review by santgr11 — APPROVED** [review]

File: libraries/botbuilder-azure/src/azureBlobTranscriptStore.ts (hunk 1 above)
Comment (ceciliaavila) [reply libraries/botbuilder-azure/src/azureBlobTranscriptStore.ts]: Feedback applied. Thanks!

File: libraries/botbuilder-azure/src/blobStorage.ts (hunk 2 above)
Comment (ceciliaavila) [reply libraries/botbuilder-azure/src/blobStorage.ts]: Feedback applied. Thanks!

File: libraries/botbuilder-azure/src/cosmosDbPartitionedStorage.ts (hunk 3 above)
Comment (ceciliaavila) [reply libraries/botbuilder-azure/src/cosmosDbPartitionedStorage.ts]: Applied. Thanks!

File: libraries/botbuilder-azure/src/cosmosDbStorage.ts (hunk 4 above)
Comment (ceciliaavila) [reply libraries/botbuilder-azure/src/cosmosDbStorage.ts]: Applied. Thanks!

**Review by sw-joelmut — APPROVED** [review]

**Review by iscaltritti-sw — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**